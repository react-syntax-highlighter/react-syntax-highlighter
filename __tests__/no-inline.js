import React from 'react';
import renderer from 'react-test-renderer';
import SyntaxHighlighter from '../src';

const code = `const woah = fun => fun + 1;
const dude = woah(2) + 3;
function thisIsAFunction() {
  return [1,2,3].map(n => n + 1).filter(n !== 3);
}
console.log('making up fake code is really hard');

function itIs() {
  return 'no seriously really it is';
}
`;

test('SyntaxHighlighter component renders correctly without inline styles when useInlineStyles=false', () => {
  const tree = renderer
    .create(
      <SyntaxHighlighter language="javascript" useInlineStyles={false}>
        {code}
      </SyntaxHighlighter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
