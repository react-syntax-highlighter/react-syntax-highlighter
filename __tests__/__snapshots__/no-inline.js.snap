// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyntaxHighlighter component renders correctly without inline styles when useInlineStyles=false 1`] = `
<pre
  className="hljs"
  style={Object {}}
>
  <code
    className="language-javascript"
    style={
      Object {
        "whiteSpace": "pre",
      }
    }
  >
    <span
      className="hljs-keyword"
    >
      const
    </span>
    <span
      className=""
    >
       woah = 
    </span>
    <span
      className="hljs-function hljs-params"
    >
      fun
    </span>
    <span
      className="hljs-function"
    >
       =&gt;
    </span>
    <span
      className=""
    >
       fun + 
    </span>
    <span
      className="hljs-number"
    >
      1
    </span>
    <span
      className=""
    >
      ;

    </span>
    <span
      className=""
    >
      
    </span>
    <span
      className="hljs-keyword"
    >
      const
    </span>
    <span
      className=""
    >
       dude = woah(
    </span>
    <span
      className="hljs-number"
    >
      2
    </span>
    <span
      className=""
    >
      ) + 
    </span>
    <span
      className="hljs-number"
    >
      3
    </span>
    <span
      className=""
    >
      ;

    </span>
    <span
      className=""
    >
      
    </span>
    <span
      className="hljs-function hljs-keyword"
    >
      function
    </span>
    <span
      className="hljs-function"
    >
       
    </span>
    <span
      className="hljs-function hljs-title"
    >
      thisIsAFunction
    </span>
    <span
      className="hljs-function"
    >
      (
    </span>
    <span
      className="hljs-function"
    >
      ) 
    </span>
    <span
      className=""
    >
      {

    </span>
    <span
      className=""
    >
        
    </span>
    <span
      className="hljs-keyword"
    >
      return
    </span>
    <span
      className=""
    >
       [
    </span>
    <span
      className="hljs-number"
    >
      1
    </span>
    <span
      className=""
    >
      ,
    </span>
    <span
      className="hljs-number"
    >
      2
    </span>
    <span
      className=""
    >
      ,
    </span>
    <span
      className="hljs-number"
    >
      3
    </span>
    <span
      className=""
    >
      ].map(
    </span>
    <span
      className="hljs-function hljs-params"
    >
      n
    </span>
    <span
      className="hljs-function"
    >
       =&gt;
    </span>
    <span
      className=""
    >
       n + 
    </span>
    <span
      className="hljs-number"
    >
      1
    </span>
    <span
      className=""
    >
      ).filter(n !== 
    </span>
    <span
      className="hljs-number"
    >
      3
    </span>
    <span
      className=""
    >
      );

    </span>
    }

    <span
      className=""
    >
      
    </span>
    <span
      className="hljs-built_in"
    >
      console
    </span>
    <span
      className=""
    >
      .log(
    </span>
    <span
      className="hljs-string"
    >
      'making up fake code is really hard'
    </span>
    <span
      className=""
    >
      );

    </span>
    

    <span
      className=""
    >
      
    </span>
    <span
      className="hljs-function hljs-keyword"
    >
      function
    </span>
    <span
      className="hljs-function"
    >
       
    </span>
    <span
      className="hljs-function hljs-title"
    >
      itIs
    </span>
    <span
      className="hljs-function"
    >
      (
    </span>
    <span
      className="hljs-function"
    >
      ) 
    </span>
    <span
      className=""
    >
      {

    </span>
    <span
      className=""
    >
        
    </span>
    <span
      className="hljs-keyword"
    >
      return
    </span>
    <span
      className=""
    >
       
    </span>
    <span
      className="hljs-string"
    >
      'no seriously really it is'
    </span>
    <span
      className=""
    >
      ;

    </span>
    }

    
  </code>
</pre>
`;
